seed/models/analysis_messages.py

```diff
@@ -0,0 +1,27 @@
+from django.db import models
+
+from seed.models import (
+    Analysis,
+    AnalysisPropertyView,
+    AnalysisTypes
+)
+
+
+class AnalysisMessage(models.Model):
+    """
+    The AnalysisMessage represents user-facing messages of events that occur
+    during an analysis, like a breadcrumb trail.
+    """
+    # if the message is relevant to the entire analysis, ie not property specific,
+    # then it should be linked to an analysis and not linked to an AnalysisPropertyView
+    # e.g. the API for the analysis service returned a non 200 response
+    analysis = models.ForeignKey(Analysis, on_delete=models.CASCADE)
+    # if the message is relevant to a specific property then it should be linked
+    # to an AnalysisPropertyView
+    # e.g. if the AnalysisPropertyView is missing some required data
+    analysis_property_view = models.ForeignKey(AnalysisPropertyView, on_delete=models.CASCADE)
```

Comment:
I'm wondering what our requirements for this should be (we didn't really specify in our diagram).

Should a message always be linked to an analysis? Or should it be either-or with a linked analysis property view? I think we should always require analysis

I'm thinking analysis_property_view should be optional.

@adrian-lara @Ryoken Thoughts?

Reply:
AnalysisPropertyView relation should definitely be optional. I'd lean towards making the Analysis relation required just so we can't ever have a hanging Message?

Reply:
In the past when there were mulitple types of parents like this we'd use a pair of fields like "parent" and "parent_type".  That way we could still enforce having to have a parent, but I'm not sure how all that works with Django.  I agree on keeping analysis required and making analasys_property_view optional, but keeping both delete triggers.

Reply:
Hmm, re: parent fields, there might be some type of polymorphism thing we could implement in Django, but might be too much work. I'm good with requiring the analysis link, and making the AnalysisPropertyView optional 👍